aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -221,6 +243,48 @@ module aptos_framework::delegation_pool {
         distribute_commission_events: EventHandle<DistributeCommissionEvent>,
     }
 
+    struct VotingRecordKey has copy, drop, store {
+        voter: address,
+        proposal_id: u64,
+    }
+
+    struct VotingPowerDelegation has copy, drop, store {
+        // The current voter address of this delegator.
+        voter: address,
+        // The change of voter address won't take effects until the current lockup period ends. Record the new voter
+        // address and update `voter` to `voter_next_lockup` when that lockup period has already ended.
+        voter_next_lockup: address,
+        // locked_until_secs of the stake pool when last time this struct is updated.
+        // If <locked_until_secs of the stake pool> != last_locked_until_secs, it means that a lockup period has passed.
+        // This is slightly different from ObservedLockupCycle because ObservedLockupCycle cannot detect if a lockup
+        // period is passed when there is no unlocking during the lockup period .
+        last_locked_until_secs: u64,
+    }
+
+    struct VoterVotingPower has copy, drop, store {
+        // Equals to the sum of active shares of delegators who delegate to this voter.
+        active_shares: u128,
+        // Equals to the sum of pending inactive shares of delegators who delegate to this voter.
+        pending_inactive_shares: u128,
+        // Equals to the sum of active shares of delegators who delegate to this voter after the current lockup period.
+        // `active_shares_next_lockup` might be different `active_shares` when some delegators change their voter.
```

Comment:
Hmm this comment is not very helpful. How about "Total active shares delegated to this voter in the next lockup cycle." I'd add more documentation regarding the fact the a voter's total delegated voting power doesn't change until the start of a new lockup cycle